classes/manage_books.rb

```diff
@@ -0,0 +1,23 @@
+class ManageBooks
+  def initialize(books)
+    @books = books
+  end
+
+  def list_books(show_index: false)
+    @books.each_with_index do |book, index|
+      result = "Title: \"#{book.title}\", Author: #{book.author}"
+      result = " #{index}) " + result if show_index
+      puts result
+    end
+  end
+
+  def add_book
+    print 'Enter the name of the book: '
+    name = gets.chomp.to_s
+    print 'Enter the author of the book: '
+    author = gets.chomp.to_s
+    book = Book.new(name, author)
+    @books << book
+    puts 'Book created successfully'
+  end
```

Comment:
- [ ] in this method you are printing to the console, getting user response and processing the response, this is not in line with the single responsibility rule, please have a look at [this article](https://github.com/microverseinc/curriculum-ruby/blob/main/oop/lessons/solid.md#best-practices-and-solid) for more information.

Reply:
Hi @mikethreels! First of all, thank you so much for the nice feedback, but I have a question in here. What can be the possible solution to split the input and output from the business logic here? you may suggest another method for printing the output in the terminal, right? 

So if I create another method just for outputting the text, I'll still pass the output text to the method as a parameter which will become the same concept. And the same concept is true about getting the input.
According to a few years of experience that I have in software development, it's not possible to split the output and input from the business logic in this method, and creating another class or method is just a mess.